src/_guides/libraries/c-interop.md

```diff
@@ -0,0 +1,142 @@
+---
+title: "C interop using FFI"
+description: "To use C code in your Dart program, use the dart:ffi library (currently in preview)."
+hw: "https://github.com/mjohnsullivan/ffi/blob/master/hello_world"
+---
+
+Dart mobile, command-line, and server apps running on the [Dart Native
+platform](/platforms/) can use the dart:ffi library to call native C APIs.
+The _ffi_ name stands for [_foreign function interface._][FFI]
+Other terms for similar functionality include _native interface_
+and _language bindings._
+
+The dart:ffi library is [in active development][ffi issue]
+and isn't complete yet.
+Although its API reference documentation hasn't been published yet,
+<!-- https://github.com/dart-lang/sdk/issues/37946 -->
+you can find its source code
+[in the SDK repo.](https://github.com/dart-lang/sdk/tree/master/sdk/lib/ffi)
+
+## Examples
+
+The following examples show how to use the dart:ffi library:
```

Comment:
@kwalrath, are these going to stay in Matt's repo, or are we moving them to this repo to make it more self contained?

Reply:
They're moving to the new samples repo: https://github.com/dart-lang/samples/pull/1